pkg/utils/logutil/log.go

```diff
@@ -88,22 +94,120 @@ func LogPanic() {
 	}
 }
 
+// RedactInfoLogType is the behavior of redacting sensitive information in logs.
+type RedactInfoLogType int
+
+const (
+	// RedactInfoLogOFF means log redaction is disabled.
+	RedactInfoLogOFF RedactInfoLogType = iota
+	// RedactInfoLogON means log redaction is enabled, and will replace the sensitive information with "?".
+	RedactInfoLogON
+	// RedactInfoLogMark means log redaction is enabled, and will use ‹› to enclose the sensitive information.
+	RedactInfoLogMark
+)
+
+// MarshalJSON implements the `json.Marshaler` interface to ensure the compatibility.
+func (t RedactInfoLogType) MarshalJSON() ([]byte, error) {
+	switch t {
+	case RedactInfoLogON:
+		return json.Marshal(true)
+	case RedactInfoLogMark:
+		return json.Marshal("MARK")
+	default:
+	}
+	return json.Marshal(false)
+}
+
+const invalidRedactInfoLogTypeErrMsg = `the "redact-info-log" value is invalid; it should be either false, true, or "MARK"`
+
+// UnmarshalJSON implements the `json.Marshaler` interface	to ensure the compatibility.
+func (t *RedactInfoLogType) UnmarshalJSON(data []byte) error {
+	var s string
+	if err := json.Unmarshal(data, &s); err == nil {
+		switch strings.ToUpper(s) {
+		case "FALSE", "OFF", "":
```

Comment:
The behavior here seems different than before, do you need to add it to the release note?

Reply:
I won't expose the "FALSE" or "OFF" (so does "TRUE" or "ON") to the user in the release note and documentation. This is only to ensure the robustness.